src/CMakeLists.txt

```diff
@@ -29,11 +29,11 @@ file(GLOB SELF_FSRC "${CMAKE_CURRENT_SOURCE_DIR}/*.f*")
 if(SELF_ENABLE_GPU)
     file(GLOB SELF_BACKEND_FSRC "${CMAKE_CURRENT_SOURCE_DIR}/gpu/*.f*")
     file(GLOB SELF_BACKEND_CPPSRC "${CMAKE_CURRENT_SOURCE_DIR}/gpu/*.cpp*")
-    if(hip_FOUND)
-      set_source_files_properties(${SELF_CPPSRC} PROPERTIES LANGUAGE HIP)
-    elseif(cuda_FOUND)
-      set_source_files_properties(${SELF_CPPSRC} PROPERTIES LANGUAGE CUDA)
-    endif()
+    # if(hip_FOUND)
```

Comment:
If these lines are commented out should they just be removed?

Reply:
Ultimately, we want to be able to use the language support for HIP/CUDA rather than bringing in HIP/CUDA through `find_package`. At the moment we are doing a hack overrided the CXX compiler with either hipcc or nvcc. 

The reason we're doing it this way (hacky) at the moment is that we get segmentation faults on our AMD GPU tests when using the HIP language support via Cmake, for some yet unknown reason. 

I'd like to keep this commented bit in to remind us to take care of this. I've added a comment in the file to this effect and will open an issue as well to remind us to look into using built in HIP and CUDA language support in CMake.

Reply:
Comment added in [3738412](https://github.com/FluidNumerics/SELF/pull/53/commits/37384125cd5dd1e4542576d1b53b8a362304e2a2)